Fix expiring Cloud Storage tokens (#15005)
decidim/decidim: RELEASE_NOTES.md

bin/rails db:migrate
```

### 1.4. AWS/Azure/Google Cloud assets storage

> As of this upgrade, you can avoid the asset caching issue due to the authentication token that expires. The rails team added an extra active storage parameter, `public: true` that you can add it to your s3 configuration in your storage.yml

Reviewer: ```suggestion
There is a bug related to the cache expiration using Active Storage (assets, such as images). For fixing this issue, the Rails team added an extra active storage parameter, `public: true` that you can add it to your storage configuration. If you followed the step `3.4. Deprecation of Rails.application.secrets` and changed your `config/storage.yml` file you don't need to do anything else.
```

```yaml
s3:
  service: S3
  public: <%= Decidim::Env.new("AWS_PUBLIC", "false").to_boolean_string %>
  access_key_id: <%= Decidim::Env.new("AWS_ACCESS_KEY_ID").to_s %>
  secret_access_key: <%= Decidim::Env.new("AWS_SECRET_ACCESS_KEY").to_s %>
  bucket: <%= Decidim::Env.new("AWS_BUCKET").to_s %>
  <%= "region: #{Decidim::Env.new("AWS_REGION").to_s}" if Decidim::Env.new("AWS_REGION").present? %>
  <%= "endpoint: #{ Decidim::Env.new("AWS_ENDPOINT").to_s}" if Decidim::Env.new("AWS_ENDPOINT").present? %>

> ```

Reviewer: ```suggestion
```

The parameter will actually strip from any asset url the `X-Amz-Credential` parameter, leaving the asset storage url like: `https://BUCKET-NAME.s3.amazonaws.com/ASSET_ID`, making it easier to cache.

To achieve that, the Content Security Policy (found in [system panel](https://docs.decidim.org/en/develop/configure/system)) comes to your aid. Populate (or append to) the following fields with the bucket url (ex: <https://decidim-bucket.s3.amazonaws.com/>): `Default src`, `Img src`, `Media src` and `Connect src`.

> Apart of that, you also need to configure your preferred cloud service provider to support this. We recommend you to follow the Rails official guide for [Active Storage configuration](https://guides.rubyonrails.org/v7.0/active_storage_overview.html#setup).

Reviewer: ```suggestion
This will also change the URL that is used, so you will need to update your [Content Security Policy](https://docs.decidim.org/en/develop/customize/content_security_policy.html), adding the new URL in the policies "default-src", "img-src", "media-src", and "connect-src". For instance, in the case of S3 with AWS, the format of the URL is the following:  `https://BUCKET-NAME.s3.amazonaws.com/ASSET_ID`.

Apart of that, you also need to configure your preferred cloud service provider to support this. We recommend you to follow the Rails official guide for [Active Storage configuration](https://guides.rubyonrails.org/v7.0/active_storage_overview.html#setup).

You can read more about this change on PR [#15005](https://github.com/decidim/decidim/pull/15005/).
```
